### ISSUE TITLE
improve search result quality

### ISSUE THREAD

**Issue by hanspagel** [scalar/scalar#2009]
### Discussed in https://github.com/scalar/scalar/discussions/1871

<div type='discussions-op-text'>

<sup>Originally posted by **rubenfiszel** May 27, 2024</sup>
One of the strength of scalar is the search function but the search work really badly in my experience.

For instance, typing: `list jobs` in https://app.windmill.dev/openapi.html#/

show 
![image](https://github.com/scalar/scalar/assets/275584/c5cd22a3-893b-4661-9df0-72ec135ec02f)

when there are actual methods that are `list all jobs`, `list queued jobs`, `list completed jobs` 
![image](https://github.com/scalar/scalar/assets/275584/ff8f8578-4d62-41aa-8c4f-4a55df9fce4c)

I'm almost sufficiently annoyed with it that I've considered doing a PR :D 

There are lots of great fuzzy search library out there, I recommend: https://github.com/leeoniya/uFuzzy</div>

**Comment by amritk** [comment]
yep fuzzy search always! Maybe I can take a look when I pull search into the new client

**Comment by thelinuxlich** [comment]
I agree, we are suffering with this now that we have a lot of endpoints

**Comment by thelinuxlich** [comment]
Also something that would probably be a easy gain is doing the search prioritizing the endpoint URL first, then other properties

**Comment by gabynevada** [comment]
The search on the client sidebar seems really bad, for some reason I can only search by http method, can't search by url portion or tag name.

**Comment by nicodand** [comment]
@amritk Are there currently any plans on your side on improving the search quality? The main pain point for us is that you can only search for endpoint summaries and not the endpoint path like "/settings/test_smtp"

**Comment by amritk** [comment]
Yea I thought @hanspagel did something with search recently. if not we can certainly improve it on the new store

**Comment by nicodand** [comment]
That would be great. Like @thelinuxlich pointed out, if the search prioritized the endpoint URL first this will already help a lot.

**Comment by hanspagel** [comment]
I think it’s working great now :)

<img width="563" height="246" alt="Image" src="https://github.com/user-attachments/assets/9970e195-aa74-48c6-91b9-2d9d393f1ab1" />

<img width="565" height="139" alt="Image" src="https://github.com/user-attachments/assets/c353d653-11b5-4c0b-8db0-97cf7b144232" />
